File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    points = [0, 0, 0]
    man = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    for index, ans in enumerate(answers):
        for i in range(3):
            if ans == man[i][index % len(man[i])]:
                points[i] += 1
    
    max_point = max(points)
    
    for i in range(3):
        if max_point == points[i]: answer.append(i + 1)
    
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    const answer = [];
    const points = [0, 0, 0];
    const man = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ];
    
    answers.forEach((answer, index) => {
        for (let i = 0; i < 3; i++) {
            if (answer === man[i][index % man[i].length]) {
                points[i] += 1;
            };
        };
    });
    
    const max_point = Math.max(...points);
    
    for (let i = 0; i < 3; i++) {
        if (max_point === points[i]) answer.push(i + 1);
    }
    
    return answer;
}